src/TeamRegistry.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+contract TeamRegistry {
+    // -------------------------------------------------------------------------
+    // Errors
+    // -------------------------------------------------------------------------
+
+    /// @notice Emitted when a member is already part of a team.
+    /// @param  member The member of the team.
+    error AlreadyInTeam(address member);
+
+    /// @notice Emitted when the caller is the team leader.
+    /// @param  teamId The ID of the team.
+    /// @param  leader The leader of the team.
+    error IsTeamLeader(uint32 teamId, address leader);
+
+    /// @notice Emitted when the user does not have a pending invite.
+    /// @param  teamId The ID of the team.
+    /// @param  member The member of the team.
+    error NoPendingInvite(uint32 teamId, address member);
+
+    /// @notice Emitted when the player is not a team member.
+    /// @param  teamId The ID of the team.
+    /// @param  player The player.
+    error NotInTeam(uint32 teamId, address player);
+
+    /// @notice Emitted when the caller is not the team leader.
+    /// @param  teamId The ID of the team.
+    /// @param  player The member of the team.
+    error NotTeamLeader(uint32 teamId, address player);
+
+    // -------------------------------------------------------------------------
+    // Events
+    // -------------------------------------------------------------------------
+
+    /// @notice Emitted when a new team is created.
+    /// @param  teamId The ID of the team.
+    /// @param  leader The leader of the team.
+    event TeamCreated(uint32 teamId, address leader);
+
+    /// @notice Emitted when a team invite has been accepted.
+    /// @param  teamId The ID of the team.
+    /// @param  member The member of the team that accepted the invite.
+    event TeamInviteAccepted(uint32 teamId, address member);
+
+    /// @notice Emitted when team leadership is transferred.
+    /// @param  oldLeader The old leader of the team.
+    /// @param  newLeader The new leader of the team.
+    event TeamLeadershipTransferred(uint32 teamId, address oldLeader, address newLeader);
```

Comment:
I've been using "present" tense for event names e.g. `AddPuzzle`, so let's keep that consistent here: `CreateTeam`, `AcceptTeamInvite`, `TransferTeamLeadership`